Nightly builds are failing due to file permission errors

Nightly builds are currently [failing](https://github.com/timberio/vector/runs/633440667?check_suite_focus=true):

```
 Complete!
Removing intermediate container 0a63f4e78a54
 ---> 7ccd823d3dc3
Successfully built 7ccd823d3dc3
Successfully tagged vector_package-rpm-x86_64:latest
Attaching to vector_package-rpm-x86_64_1
package-rpm-x86_64_1                              | Packaging .rpm for vector-x86_64-unknown-linux-musl.tar.gz
package-rpm-x86_64_1                              | TARGET: x86_64-unknown-linux-musl
package-rpm-x86_64_1                              | rm: cannot remove '/root/rpmbuild/SOURCES': Permission denied
vector_package-rpm-x86_64_1 exited with code 1

Makefile:177: recipe for target 'package-rpm-x86_64' failed
##[error]Process completed with exit code 2.
```

This is likely due to https://github.com/timberio/vector/pull/2486. I'm curious if we need to make changes in our nightly workflow?

--- a/distribution/rpm/vector.spec
+++ b/distribution/rpm/vector.spec
@@ -25,9 +25,6 @@ URL: %{_url}
 tar -xvf %{_sourcedir}/%{_source} --strip-components=2
 cp -a %{_sourcedir}/systemd/. systemd
 
-chown -R root.root .
-chmod -R a+rX,g-w,o-w .
-
 %install
 # We are currently in the BUILDROOT dir
 rm -rf %{buildroot}

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -59,10 +59,14 @@ ARCH=${ARCH:-$(echo $TARGET | cut -d'-' -f1)}
 # Prepare rpmbuild dir
 RPMBUILD_DIR="$(mktemp -td "rpmbuild.XXXX")"
 
-# Create source dir
-rm -rf "$RPMBUILD_DIR/SOURCES"
+# Create build dirs
+for ITEM in RPMS SOURCES SPECS SRPMS BUILD; do
+  rm -rf "${RPMBUILD_DIR:?}/${ITEM:?}"
+  mkdir -p "$RPMBUILD_DIR/$ITEM"
+done
+
+# Init support data
 mkdir -p \
-  "$RPMBUILD_DIR/SOURCES" \
   "$RPMBUILD_DIR/SOURCES/init.d" \
   "$RPMBUILD_DIR/SOURCES/systemd"
 cp -av distribution/init.d/. "$RPMBUILD_DIR/SOURCES/init.d"

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -68,7 +68,7 @@ cp -av distribution/systemd/. "$HOME/rpmbuild/SOURCES/systemd"
 cp -av $archive_path "$HOME/rpmbuild/SOURCES/vector-$ARCH.tar.gz"
 
 # Perform the build.
-rpmbuild --target "$ARCH-redhat-linux" --define "_arch $ARCH" -ba distribution/rpm/vector.spec
+rpmbuild  --define "_topdir $HOME/rpmbuild" --target "$ARCH-redhat-linux" --define "_arch $ARCH" -ba distribution/rpm/vector.spec
 
 #
 # Move the RPM into the artifacts dir

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -29,6 +29,15 @@ echo "Packaging .rpm for $archive_name"
 echo "TARGET: $TARGET"
 
 #
+# Safeguard
+#
+
+if [[ "$UID" == "0" ]]; then
+  echo "Error: aborting RPM build due to execution as root" >&2
+  exit 1
+fi
+
+#
 # Package
 #
 

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -57,7 +57,7 @@ CLEANED_VERSION=$(echo $CLEANED_VERSION | sed 's/-/\./g')
 ARCH=${ARCH:-$(echo $TARGET | cut -d'-' -f1)}
 
 # Prepare rpmbuild dir
-RPMBUILD_DIR="$HOME/rpmbuild"
+RPMBUILD_DIR="$(mktemp -td "rpmbuild.XXXX")"
 
 # Create source dir
 rm -rf "$RPMBUILD_DIR/SOURCES"

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -48,15 +48,15 @@ CLEANED_VERSION=$(echo $CLEANED_VERSION | sed 's/-/\./g')
 ARCH=${ARCH:-$(echo $TARGET | cut -d'-' -f1)}
 
 # Create source dir
-rm -rf /root/rpmbuild/SOURCES
-mkdir -p /root/rpmbuild/SOURCES
-mkdir -p /root/rpmbuild/SOURCES/init.d
-mkdir -p /root/rpmbuild/SOURCES/systemd
-cp -av distribution/init.d/. /root/rpmbuild/SOURCES/init.d
-cp -av distribution/systemd/. /root/rpmbuild/SOURCES/systemd
+rm -rf "$HOME/rpmbuild/SOURCES"
+mkdir -p "$HOME/rpmbuild/SOURCES"
+mkdir -p "$HOME/rpmbuild/SOURCES/init.d"
+mkdir -p "$HOME/rpmbuild/SOURCES/systemd"
+cp -av distribution/init.d/. "$HOME/rpmbuild/SOURCES/init.d"
+cp -av distribution/systemd/. "$HOME/rpmbuild/SOURCES/systemd"
 
 # Copy the archive into the sources dir
-cp -av $archive_path "/root/rpmbuild/SOURCES/vector-$ARCH.tar.gz"
+cp -av $archive_path "$HOME/rpmbuild/SOURCES/vector-$ARCH.tar.gz"
 
 # Perform the build.
 rpmbuild --target "$ARCH-redhat-linux" --define "_arch $ARCH" -ba distribution/rpm/vector.spec
@@ -65,5 +65,5 @@ rpmbuild --target "$ARCH-redhat-linux" --define "_arch $ARCH" -ba distribution/r
 # Move the RPM into the artifacts dir
 #
 
-ls "/root/rpmbuild/RPMS/$ARCH"
-mv -v "/root/rpmbuild/RPMS/$ARCH/vector-$CLEANED_VERSION-$RELEASE.$ARCH.rpm" "target/artifacts/vector-$ARCH.rpm"
+ls "$HOME/rpmbuild/RPMS/$ARCH"
+mv -v "$HOME/rpmbuild/RPMS/$ARCH/vector-$CLEANED_VERSION-$RELEASE.$ARCH.rpm" "target/artifacts/vector-$ARCH.rpm"

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -56,21 +56,24 @@ CLEANED_VERSION=$(echo $CLEANED_VERSION | sed 's/-/\./g')
 # from Rust target triple and needs to be specified manually.
 ARCH=${ARCH:-$(echo $TARGET | cut -d'-' -f1)}
 
+# Prepare rpmbuild dir
+RPMBUILD_DIR="$HOME/rpmbuild"
+
 # Create source dir
-rm -rf "$HOME/rpmbuild/SOURCES"
+rm -rf "$RPMBUILD_DIR/SOURCES"
 mkdir -p \
-  "$HOME/rpmbuild/SOURCES" \
-  "$HOME/rpmbuild/SOURCES/init.d" \
-  "$HOME/rpmbuild/SOURCES/systemd"
-cp -av distribution/init.d/. "$HOME/rpmbuild/SOURCES/init.d"
-cp -av distribution/systemd/. "$HOME/rpmbuild/SOURCES/systemd"
+  "$RPMBUILD_DIR/SOURCES" \
+  "$RPMBUILD_DIR/SOURCES/init.d" \
+  "$RPMBUILD_DIR/SOURCES/systemd"
+cp -av distribution/init.d/. "$RPMBUILD_DIR/SOURCES/init.d"
+cp -av distribution/systemd/. "$RPMBUILD_DIR/SOURCES/systemd"
 
 # Copy the archive into the sources dir
-cp -av $archive_path "$HOME/rpmbuild/SOURCES/vector-$ARCH.tar.gz"
+cp -av $archive_path "$RPMBUILD_DIR/SOURCES/vector-$ARCH.tar.gz"
 
 # Perform the build.
 rpmbuild \
-  --define "_topdir $HOME/rpmbuild" \
+  --define "_topdir $RPMBUILD_DIR" \
   --target "$ARCH-redhat-linux" \
   --define "_arch $ARCH" \
   -ba distribution/rpm/vector.spec
@@ -79,5 +82,5 @@ rpmbuild \
 # Move the RPM into the artifacts dir
 #
 
-ls "$HOME/rpmbuild/RPMS/$ARCH"
-mv -v "$HOME/rpmbuild/RPMS/$ARCH/vector-$CLEANED_VERSION-$RELEASE.$ARCH.rpm" "target/artifacts/vector-$ARCH.rpm"
+ls "$RPMBUILD_DIR/RPMS/$ARCH"
+mv -v "$RPMBUILD_DIR/RPMS/$ARCH/vector-$CLEANED_VERSION-$RELEASE.$ARCH.rpm" "target/artifacts/vector-$ARCH.rpm"

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -58,9 +58,10 @@ ARCH=${ARCH:-$(echo $TARGET | cut -d'-' -f1)}
 
 # Create source dir
 rm -rf "$HOME/rpmbuild/SOURCES"
-mkdir -p "$HOME/rpmbuild/SOURCES"
-mkdir -p "$HOME/rpmbuild/SOURCES/init.d"
-mkdir -p "$HOME/rpmbuild/SOURCES/systemd"
+mkdir -p \
+  "$HOME/rpmbuild/SOURCES" \
+  "$HOME/rpmbuild/SOURCES/init.d" \
+  "$HOME/rpmbuild/SOURCES/systemd"
 cp -av distribution/init.d/. "$HOME/rpmbuild/SOURCES/init.d"
 cp -av distribution/systemd/. "$HOME/rpmbuild/SOURCES/systemd"
 

--- a/scripts/package-rpm.sh
+++ b/scripts/package-rpm.sh
@@ -68,7 +68,11 @@ cp -av distribution/systemd/. "$HOME/rpmbuild/SOURCES/systemd"
 cp -av $archive_path "$HOME/rpmbuild/SOURCES/vector-$ARCH.tar.gz"
 
 # Perform the build.
-rpmbuild  --define "_topdir $HOME/rpmbuild" --target "$ARCH-redhat-linux" --define "_arch $ARCH" -ba distribution/rpm/vector.spec
+rpmbuild \
+  --define "_topdir $HOME/rpmbuild" \
+  --target "$ARCH-redhat-linux" \
+  --define "_arch $ARCH" \
+  -ba distribution/rpm/vector.spec
 
 #
 # Move the RPM into the artifacts dir

